### PR TITLE
Feature/optimize checkout

### PR DESCRIPTION
**Pull request by oa1802**
## Description of the change

> Optimizing java spring vulny test repo checkout

replace me

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> [ISSUE-ID](https://<ORG>.atlassian.net/browse/ISSUE-ID)

### Development

- [x] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: .github/workflows/hawkscan-integration.yml [hunk 1]
```diff
@@ -12,10 +12,11 @@ jobs:
...
       - name: Create app directory for ${{ github.event.client_payload.repoName }}
         run: mkdir -p __tests__/integration-tests/apps
-      - name: Checkout ${{ github.event.client_payload.repoName }} repository
-        run: |
-          cd __tests__/integration-tests/apps/
-          git clone https://github.com/${{ github.event.client_payload.repoOrg }}/${{ github.event.client_payload.repoName }}.git
+      - name: Checkout javaspringvulny repository
+        uses: actions/checkout@v2
+        with:
+          repository: kaakaww/javaspringvulny
```
Comment (d-co-white) [inline .github/workflows/hawkscan-integration.yml:18]:
```suggestion
          repository: ${{ github.event.client_payload.repoOrg }}/${{ github.event.client_payload.repoName }}
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -80,6 +80,36 @@ jobs:
...
+          docker-compose up -d
+      - name: Run HawkScan
+        id: run-hawkscan
+        uses: stackhawk/hawkscan-action@main
+        with:
+          apiKey: ${{ secrets.SHAWK_API_KEY }}
+          workspace: ${{ github.workspace }}/__tests__/integration-tests/configs/javaspringvulny/
+          configurationFiles: stackhawk-jsv-json-token.yml
+          sourceURL: https://download.stackhawk.com/hawk/cli
+          version: latest
```
Comment (zconger) [inline .github/workflows/test.yml:107]: These are the defaults - testing them on purpose I presume?

File: .github/workflows/hawkscan-integration.yml [hunk 3]
```diff
@@ -12,10 +12,11 @@ jobs:
...
+          path: ./__tests__/integration-tests/apps/${{ github.event.client_payload.repoName }}
```
Comment (zconger) [inline .github/workflows/hawkscan-integration.yml:19]: Rad

**Review by zconger — DISMISSED** [review]
LGTM!

File: .github/workflows/hawkscan-integration.yml (hunk 3 above)
Comment (zconger) [reply .github/workflows/hawkscan-integration.yml:19]: Oh snap Dana has a good suggestion here.

File: .github/workflows/hawkscan-integration.yml [hunk 4]
```diff
@@ -12,10 +12,11 @@ jobs:
           ref: ${{ github.event.client_payload.ref }}
       - name: Create app directory for ${{ github.event.client_payload.repoName }}
```
Comment (d-co-white) [inline .github/workflows/hawkscan-integration.yml:13]: I think you can skip this step.  When I was testing, the checkout created the apps directory under __tests__/integration-tests

File: .github/workflows/test.yml (hunk 2 above)
Comment (oa1802) [reply .github/workflows/test.yml]: I removed these

File: .github/workflows/hawkscan-integration.yml (hunk 4 above)
Comment (oa1802) [reply .github/workflows/hawkscan-integration.yml:13]: Going to leave this for now to keep the app and config directories, but maybe will change in the future

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -80,6 +80,36 @@ jobs:
             configurationFiles: __tests__/stackhawk.yml
             codeScanningAlerts: true
 
+  javaspringvulny-test:
+    name: Java Spring Vulny Test
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Create app directory for javaspringvulny
```
[the remark this answers is not in the document]
Comment (oa1802) [reply .github/workflows/test.yml:88]: Same reply as previous comment

**Pull request merged**